src/wallet/rpcwallet.cpp

```diff
@@ -4773,6 +4773,31 @@ size_t EstimateTxSize(
     return txsize;
 }
 
+static std::optional<Memo> ParseMemo(const UniValue& memoValue)
+{
+    if (memoValue.isNull()) {
+        return std::nullopt;
+    } else {
+        return examine(Memo::FromHex(memoValue.get_str()), match {
+            [](MemoError err) -> Memo {
+                switch (err) {
+                    case MemoError::HexDecodeError:
+                        throw JSONRPCError(
+                                RPC_INVALID_PARAMETER,
+                                "Invalid parameter, expected memo data in hexadecimal format.");
+                    case MemoError::MemoTooLong:
+                        throw JSONRPCError(
+                                RPC_INVALID_PARAMETER,
+                                strprintf("Invalid parameter, size of memo is larger than maximum allowed %d", ZC_MEMO_SIZE ));
```

Comment:
For some reason this case is not being reached, and instead we now get a different error being thrown in the Boost tests:
```
test/test_util.cpp(119): error: in "rpc_wallet_tests/rpc_z_mergetoaddress_parameters": check expectedErrorMessage == e.what() has failed [Invalid parameter, memo is longer than the maximum allowed 512 characters. != Invalid parameter, size of memo is larger than maximum allowed 512]
```

Reply:
Yeah … there are like three places we handle `MemoError` with slightly different messages. My guess is that this now _is_ the case we’re hitting, and previously it was the slightly different one in `Memo::FromHexOrThrow` (which is also a `runtime_error` instead of a `JSONRPCError`.

I’ll fix the tests.

(I also have a branch on the verge of being a PR that encapsulates `Memo` better, eliminating a lot of this duplication. I was actually cleaning it up to push before I discovered the issue in this PR.)